### DISCUSSION TITLE
how to add <!DOCTYPE html> with return buildHtml:

### DISCUSSION THREAD

**Discussion by jerrygzy** [HapticX/happyx#188] [Q&A] [answered]
Hi, my code does not include `<!DOCTYPE html>`,  how to add it?

```nim
    return buildHtml:
        tHtml("data-theme" = "light"):
            {header(title)}
        tBody():
```

**Comment by Ethosa** [comment]
I don't think that you should use `<!DOCTYPE html>` in modern browsers, but if you want:
```nim
buildHtml:
  rawHtml: "<!DOCTYPE html>"
```

**Reply by jerrygzy** [reply]
according to many source from internet, 

> HTML5 Standard: `<!DOCTYPE html>` is the DOCTYPE declaration for HTML5, which is the current standard for HTML. It tells the browser to render the page in standards mode.

> 
> Always Prefer Standards Mode: For any regular web development, always use `<!DOCTYPE html>` to ensure your pages are rendered in standards mode.
> Avoid Quirks Mode: Modern web development should avoid quirks mode as it leads to inconsistent behavior across different browsers and can cause numerous issues with page rendering and CSS styling.
> Adhere to Modern Practices: Standards have evolved to make web development more consistent and accessible. Sticking to modern practices, including using `<!DOCTYPE html>`, is generally the best approach.

**Reply by jerrygzy** [reply]
btw, the solution provided caused the page render failed, could you provide a alternative method or a happyx upgrade?

**Reply by Ethosa** [reply, marked as answer]
since current commit you can use
```nim
buildHtml:
  !DOCTYPE html
```
